src/main/kotlin/com/hibob/types/StoreService.kt

```diff
@@ -0,0 +1,65 @@
+package com.hibob.types
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class StoreService {
+
+    fun checkProductCustom(custom: Any): Boolean{
+        (custom as? Boolean)?.let{
+            when(custom){
+                true -> return true
+                else -> return false
+            }
+        } ?: return false
```

Comment:
nice use of as? 
regarding to the when, in general if you have only 2 options is better use if.
in this case you can do the following:
`  return (custom as? Boolean) ?: false`